libkbfs/prefetcher.go

```diff
@@ -496,6 +586,19 @@ func (p *blockPrefetcher) run(testSyncCh <-chan struct{}) {
 			p.log.Debug("canceling prefetch for block %s", blockID)
 			// Walk up the block tree and delete every parent.
 			p.applyToParentsRecursive(p.cancelPrefetch, blockID, pre)
+		case reqInt := <-p.prefetchRescheduleCh.Out():
+			req := reqInt.(*prefetchRequest)
+			blockID := req.ptr.ID
+			pre, ok := p.prefetches[blockID]
+			if !ok {
```

Comment:
Comment here that you need to create and cache a new prefetch to stop other causes from creating a new one as well, while this one is waiting.

Also please use `isPrefetchWaiting` instead of `ok`, to stay consistent with the rest of the prefetcher.